docs/proposals/gateway.md

```diff
@@ -0,0 +1,234 @@
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: metadata.namespace
+          - name: INSTANCE_IP
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: status.podIP
+          - name: ISTIO_META_POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          volumeMounts:
+          - name: kubeedge-certs
+            mountPath: /etc/certs
+            readOnly: true
+          - name: edgegateway-certs
+            mountPath: "/etc/kubeedge/edgegateway-certs"
+            readOnly: true
+          - name: edgegateway-ca-certs
+            mountPath: "/etc/kubeedge/edgegateway-ca-certs"
+            readOnly: true
+      volumes:
+      - name: edgegateway-certs
+        secret:
+          secretName: "istio-edgegateway-certs"
+          optional: true
+      - name: edgegateway-ca-certs
+        secret:
+          secretName: "istio-edgegateway-ca-certs"
+          optional: true
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - amd64
+                - ppc64le
+                - s390x
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 2
+            preference:
+              matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - amd64
+          - weight: 2
+            preference:
+              matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - ppc64le
+          - weight: 2
+            preference:
+              matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                - s390x
+```
+
+
+
+2. User define Gateway to represent an ingress router rule to the selected EdgeGateway as well as VirtualService to represent a backend service dispatching rule. For example:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: bookinfo-gateway
+spec:
+  selector:
+    istio: edgegateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: bookinfo
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - bookinfo-gateway
+  http:
+  - match:
+    - uri:
+        exact: /productpage
+    - uri:
+        exact: /login
+    - uri:
+        exact: /logout
+    - uri:
+        prefix: /api/v1/products
+    route:
+    - destination:
+        host: productpage
+        port:
+          number: 9080

```

Comment:
If the pods of a service runs in different available areas, can this rule restrict requests to the corresponding zones and it seems that `productpage` is a service here.

Reply:
`productpage` is short name, instead of `productpage.default.svc.cluster.local`

Reply:
the destination of ingress controller is also service, so what's the difference with ingress rule?

Reply:
check [VirtualService](https://istio.io/docs/reference/config/networking/virtual-service/), they provide lots of communication governance capabilities, such as fault injection, canary release and so on. With those capabilities, the VirtualService won't be the definition of the request forward only anymore. We are planning to add those communication governance also in the future.

Reply:
Sounds good!